dev/com.ibm.ws.transaction.cdi/src/com/ibm/tx/jta/cdi/TransactionContextExtension.java

```diff
@@ -26,9 +44,83 @@
            property = { "bean.defining.annotations=javax.transaction.TransactionScoped" })
 public class TransactionContextExtension implements Extension, WebSphereCDIExtension {
 
+    private static final TraceComponent tc = Tr.register(TransactionContext.class, TranConstants.TRACE_GROUP, TranConstants.NLS_FILE);
+
+    //This is not actually used since weld will create a new instance of this class seperate from the one OSGI has populated. 
+    //But this stays so that OSGI will manage the extensions lifecycle. 
+    private final AtomicServiceReference<CDIService> cdiServiceRef = new AtomicServiceReference<CDIService>("cdiService");
+
+    private BeanManager beanManager = null;
+
     public void afterBeanDiscovery(@Observes AfterBeanDiscovery event, BeanManager manager) {
-        TransactionContext tc = new TransactionContext();
+        TransactionContext tc = new TransactionContext(getBeanManager());
         com.ibm.tx.jta.impl.TransactionImpl.registerTransactionScopeDestroyer(tc);
-        event.addContext(new TransactionContext());
+        event.addContext(tc);
+    }
+
+
+    private BeanManager getBeanManager() {
+        if (beanManager == null) {
+            BundleContext bundleContext = getBundleContext(FrameworkUtil.getBundle(getClass()));
+            if (bundleContext != null) {
+                ServiceReference<CDIService> serviceReference = getServiceReferences(bundleContext);
```

Comment:
Instead of calling `getServiceReferences`, you should be able to just use `getSerivceReference`.  See `com.ibm.ws.beanvalidation.v20.cdi.internal.LibertyHibernateValidatorExtension.getClassLoadingService()` for a similar use-case where we get the `ClassLoadingService`